Builder/AbstractFormContractor.php

```diff
@@ -53,97 +56,127 @@ public function __construct(FormFactoryInterface $formFactory)
+     *
+     * @throws MissingTargetEntityException if the field description has no target entity
+     */
+    final protected function ensureTargetEntity(FieldDescriptionInterface $fieldDescription)
+    {
+        if (!$fieldDescription->getTargetEntity()) {
+            throw $this->createMissingTargetEntityException($fieldDescription);
+        }
+    }
+
+    /**
+     * Ensures that the given field description has an association admin.
+     *
+     * @param FieldDescriptionInterface $fieldDescription
+     *
+     * @throws MissingAssociationAdminException if the field description has no association admin
+     */
+    final protected function ensureAssociationAdmin(FieldDescriptionInterface $fieldDescription)
+    {
+        if (!$fieldDescription->getAssociationAdmin()) {
+            throw $this->createMissingAssociationAdminException($fieldDescription);
+        }
+    }
+
+    /**
+     * Ensures a field description describes an association type that can be handled by the given form type.
+     *
+     * @param FieldDescriptionInterface $fieldDescription The field description to check.
+     * @param string                    $type             The association type to ensure.
+     * @param string                    $formType         The requested form type.
+     *
+     * @throws UnhandledAssociationTypeException if the field description has no association admin
+     */
+    final protected function ensureAssociationType(FieldDescriptionInterface $fieldDescription, $type, $formType)
+    {
+        if (!in_array($type, array('single-valued', 'collection-valued'), true)) {
+            throw new \InvalidArgumentException(sprintf(
+                'Second argument to %s must be either "single-valued" or "collection-valued"',
+                __METHOD__
+            ));
+        }
+        if ($type === 'single-valued' && !$fieldDescription->describesSingleValuedAssociation()) {
+            throw $this->createUnhandledAssociationTypeException($formType, $type, $fieldDescription, array(
+                'Sonata\CoreBundle\Form\Type\CollectionType',
+                'Sonata\AdminBundle\Form\Type\ModelType',
+                'Sonata\AdminBundle\Form\Type\ModelAutocompleteType',
+            ));
+        } elseif ($type === 'collection-valued' && !$fieldDescription->describesCollectionValuedAssociation()) {
+            throw $this->createUnhandledAssociationTypeException($formType, $type, $fieldDescription, array(
+                'Sonata\AdminBundle\Form\Type\AdminType',
+                'Sonata\AdminBundle\Form\Type\ModelType',
+                'Sonata\AdminBundle\Form\Type\ModelTypeList',
+                'Sonata\AdminBundle\Form\Type\ModelAutocompleteType',
+                'Sonata\AdminBundle\Form\Type\ModelHiddenType',
+                'Sonata\AdminBundle\Form\Type\ModelReferenceType',
+            ));
+        }
+    }
+
+    /**
+     * Returns a `LogicException` instance with a developer-friendly message,
```

Comment:
Markdown can be used in docblocks descriptions. I don't think it can in summaries, though it will likely depend on the tool used to generate the html...